### ISSUE TITLE
Update classes to remove Redis name

### ISSUE THREAD

**Issue by jzonthemtn** [philterd/phileas#164]
Phileas uses Redis for caching in various places - alerts, anonymization, span disambiguation, and policies.

While Phileas does not use Redis itself, these classes have names that include Redis, such as `RedisAlertService`, and its underlying `AbstractRedisCacheService`. 

Phileas does have a dependency on the [redisson](https://github.com/redisson/redisson) client library which is available under the Apache license. Redisson also supports valkey, the Redis fork available under an [open source license](https://github.com/valkey-io/valkey?tab=License-1-ov-file#readme). [AWS is offering valkey as a managed service.](https://aws.amazon.com/about-aws/whats-new/2024/10/amazon-elasticache-valkey/)

The class naming with respect to Redis should be made generic to not imply that Redis is required for Phileas, when valkey is also an option.

**Comment by jzonthemtn** [comment]
The bigger idea here is that the cache should not have to be Redis (or Valkey). The abstract cache service should allow for a lot of implementations. Having the abstract class service named `AbstractRedisCacheService` is very restricting.

**Comment by jzonthemtn** [comment]
This is no longer needed here after #202.
